src/cloudai/configurator/trajectory.py

```
@@ -0,0 +1,364 @@
+    file_name = "trajectory.csv"
+
+    def __init__(self, iteration_dir: Path | Callable[[], Path]) -> None:
+        super().__init__(iteration_dir)
+        self._fields: tuple[str, ...] | None = None
+
+    def append(self, record: Mapping[str, object]) -> None:
+        fields = tuple(record)
+        if self._fields is None:
+            self._fields = fields
+        elif fields != self._fields:
+            raise ValueError(f"trajectory record fields changed: expected {self._fields}, got {fields}")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)